nimgame2/utils.nim

```diff
@@ -551,3 +551,27 @@ proc update*(counter: Counter) =
 proc value*(counter: Counter): int {.inline.} =
   counter.current
 
+#===========#
```

Comment:
I feel that this section belongs into `types` module.

What do you think about merging last three procs into one, e.g.:

```
proc `+=`*(transform: Transform, pos: Coord = (0.0, 0.0), angle: Angle = 0.0, scale: Scale = 0.0): Transform =
  ...
```
?

Reply:
Scaling isn't `+` its `*`
The whole idea is to have a temporary transform. Something like. "What would be my transform if I was moved 5 steps forward and turned 12 degrees: `transform.translated((5.0,0.0)).rotated(12.0)`

Reply:
The reason its in utils is because of `rotate`

Reply:
With such procedure chaining you are copying the transform tuple each time though, it might hit the performance…

May be something akin to
```
proc transformed*(target: Transform, translated: Coord = (0.0, 0.0), rotated: Angle = 0.0, scaled: Scale = 1.0): Transform =
  ...

target.transformed(translated = (5.0, 0.0), rotated = (12.0))
```
then?